### DISCUSSION TITLE
Is it possible to get a function pointer from a FunctionValue, or at least not the name?

### DISCUSSION THREAD

**Discussion by PartyWumpus** [TheDan64/inkwell#547] [Q&A]
**Describe the Bug**
I'm writing a simple jit compiler using inkwell, and I'm caching functions, so I need access to the generated function pointers. If I didn't need to cache, I could just use ExecutionEngine's [`run_function`](https://thedan64.github.io/inkwell/inkwell/execution_engine/struct.ExecutionEngine.html#method.run_function) which takes in a FunctionValue. 

Seemingly the only way to access the function pointer myself is with ExectionEngine's [`get_function`](https://thedan64.github.io/inkwell/inkwell/execution_engine/struct.ExecutionEngine.html#method.get_function), which takes in a *name* which has to be globally unique (as in if 2 modules have functions with the same name there is no way to specify which to pick). Also afaict it is slower (not profiled it properly)?

Am I missing a way to get from a FunctionValue to a function pointer (specifically a JitFunction but a raw function pointer is good too i think)? I could just store the FunctionValue instead of the function pointer, but I'd rather just have a pointer directly to the function instead of going through LLVM for every call.

Sorry if this is more of a question/feature request than a bug report, but I wasn't really sure what to label it as.

**To Reproduce**
This is what I've been doing as a temporary solution, but it does not spark joy
```rust
let void_type = self.context.void_type();
let fn_type = void_type.fn_type(&[], true);

// AAAAAA
let func_name = Alphanumeric.sample_string(&mut rand::thread_rng(), 16);
let function = module.add_function(&func_name, fn_type, None);
let basic_block = self.context.append_basic_block(function, "entry");
self.builder.position_at_end(basic_block);

/// ... snip, return is built at some point

unsafe { self.execution_engine.get_function(&func_name)? }
```

**LLVM Version (please complete the following information):**
 - LLVM Version: 18.1.8
 - Inkwell Branch Used: llvm18-0

**Desktop (please complete the following information):**
 - OS: NixOS 24.11

**Comment by TheDan64** [comment]
Converted to a discussion as it is more of a question than a feature or bug

**Reply by PartyWumpus** [reply]
Ahh discussions, yep that's what I should've done. Thanks.
